src/blog/2024/05/mapping-location-on-dashboard-2.md

```diff
@@ -0,0 +1,130 @@
+   - location
+---
+
+Have you ever needed to develop a fleet management system or something similar that requires a map with plotted location data on a dashboard? If so, this guide is for you. In this guide, we will learn how to plot location data on a world map within Dashboard 2.0.
+
+<!--more-->
+
+Before moving further, make sure you have installed Dashboard 2.0. If you are new to Dashboard 2.0, please refer to [Getting Started with Dashboard 2.0](/blog/2024/03/dashboard-getting-started/) for more information.
+
+## Installing world map custom node
+
+To render an interactive world map webpage for plotting location data, we will use a popular custom node called [node-red-contrib-web-worldmap-page](https://flows.nodered.org/node/node-red-contrib-web-worldmap). This node offers extensive features enabling us to render a world map and plot various items with different icons, colors, NATO symbologies, ranges, and more.
+
+1. Click the Node-RED Settings (top-right).
+2. Click "Manage Palette".
+3. Switch to the "Install" tab.
+4. Search for `node-red-contrib-web-worldmap`.
+5. Click "Install".
+
+## Retrieving location Data
+
+Before plotting locations, we need to obtain the data first. For this purpose, we will utilize the [Edenburg Open Public Transportation API](https://tfe-opendata.readme.io/docs/getting-started). This API provides the live locations of all of Edenburg's public buses and trams, enabling us to access the necessary data for plotting on Worldmap within Dashboard 2.0.
+
+1. Drag an **Inject** node onto the canvas and set the repeat property to a 20-second interval.
+2. Drag an HTTP request node onto the canvas, Dobule-click on it and choose **GET** method, and enter `https://tfe-opendata.com/api/v1/vehicle_locations` in the URL field and select return  as **a parsed json object**. This API is public, so no need for environment variables. For private APIs, consider using [environment variables](/blog/2023/01/environment-variables-in-node-red) for security.
+
+!["Screenshot of the HTTP request node configuration for retrieving data from the API"](./images/mapping-location-on-dashboard-2-http-request-node.png "Screenshot of the HTTP request node configuration for retrieving data from the API"){data-zoomable}
+
+3. Connect the **Inject** node's output to the **HTTP request** node's input.
+
+## Formatting Location Data
+
+To ensure compatibility with the Worldmap custom node, we need to format the location data appropriately. 
+
+1. Drag the **Change** node onto the canvas, and set the `msg.payload` to `msg.payload.vehicles`.
+
+!["Screenshot of the Change node setting the payload to the vehicles JSON array containing actual vehicle location data"](./images/mapping-location-on-dashboard-2-change-node.png "Screenshot of the Change node setting the payload to the vehicles JSON array containing actual vehicle location data"){data-zoomable}
+
+2. Drag a **Function** node onto the canvas and paste the following code into it.
```

Comment:
No, this is a split and join node sandwiching some other nodes. Among which a change node. Please keep things low code, youre not showing off your JS knowledge but how capable you are keeping things close to no code

Reply:
@sumitshinde-84 can you add an example to the split core node doc that explains how to handle mapping a collection?

Reply:
Yes Zj, I will !

Reply:
> @sumitshinde-84 can you add an example to the split core node doc that explains how to handle mapping a collection?

I have now updated the blog and flow to use core nodes.